backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/types/predicate/impl/ElasticsearchTextFieldPredicateBuilderFactory.java

```diff
@@ -42,7 +44,9 @@ public boolean hasCompatibleAnalyzer(ElasticsearchFieldPredicateBuilderFactory o
 		}
 
 		ElasticsearchTextFieldPredicateBuilderFactory castedOther = (ElasticsearchTextFieldPredicateBuilderFactory) other;
-		return Objects.equals( analyzer, castedOther.analyzer ) && Objects.equals( normalizer, castedOther.normalizer );
+		return Objects.equals( analyzer, castedOther.analyzer ) &&
+				Objects.equals( searchAnalyzer, castedOther.searchAnalyzer ) &&
```

Comment:
If the search analyzers are the same but the analyzers are different, this should return true, I think. And the tests should probably reflect that... ?
In fact, you could solve this easily by only keeping the search analyzer in this class, and using the analyzer as a default when the search analyzer is `null`.

Reply:
You're right!